src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -958,6 +964,13 @@ protected internal virtual void Renew()
 
         protected internal virtual unsafe void Reset(bool forceReleaseLongLivedAllocator = false)
         {
+            //TODO: Maybe not clearing allocation since we are going to reset the arena anyway and this is redundent and costly?
```

Comment:
I think that this is a good idea, yes

Reply:
If this is the case then we don't really need to do anything but generate a new modifier per builder.
Maybe we should just remove the interface generate a new modifier and that's all.

Reply:
modifying it to a struct breaks the code since it is passed as a parameter

Reply:
A technique similar to what I do on the Allocators branch could probably works.